components/Buttons/src/MDCFloatingButton+Animation.m

```diff
@@ -0,0 +1,92 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCFloatingButton+Animation.h"
+
+static const CGFloat kMDCFloatingButtonAnimationDuration = 0.18F;
+
+@implementation MDCFloatingButton (Animation)
+
+- (void)grow:(BOOL)animated completion:(void (^_Nullable)(void))completion {
+  void (^growActions)() = ^{
+    self.transform = CGAffineTransformIdentity;
+  };
+  if (animated) {
+    [UIView animateWithDuration:kMDCFloatingButtonAnimationDuration
+                     animations:^{
```

Comment:
This probably needs a curve / motion design. Did it get vetted by a motion designer?